### PR TITLE
Jack2 netsource.c

### PR DESCRIPTION
**Pull request by dvzrv**
Add the jack2 variant of tools/netsource.c, as it had been imported to a different location (with added modifications), which makes cherry-picking or rebase impossible.

Relates to #9

### CODE REVIEW

File: tools/netsource.c [hunk 1]
```diff
@@ -341,97 +373,78 @@ process (jack_nframes_t nframes, void *arg)
...
-	input_fd = outsockfd;
+        input_fd = outsockfd;
 
     // for latency == 0 we can poll.
-    if( (latency == 0) || (freewheeling!=0)  ) {
-	jack_time_t deadline = jack_get_time() + 1000000 * jack_get_buffer_size(client)/jack_get_sample_rate(client);
-	// Now loop until we get the right packet.
-	while(1) {
-		jack_nframes_t got_frame;
-#ifdef __JACK1__
```
Comment (falkTX) [inline tools/netsource.c:354]: the `__JACK1__` and `__JACK2__` macros are being removed here, is this intended?

Comment (dvzrv) [reply tools/netsource.c]: Cherry-picked and adapted 7a8db80a703543b12097f9c32048b7784a69363c

File: tools/netsource.c [hunk 2]
```diff
@@ -36,6 +36,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 #ifdef WIN32
 #include <winsock2.h>
+#define socklen_t int
```
Comment (falkTX) [inline tools/netsource.c:39]: ouch was this one here before? it is nasty to define types like that.

File: tools/netsource.c (hunk 1 above)
Comment (dvzrv) [reply tools/netsource.c:354]: For posterity: I have removed that commit again, as it turns out that we confused one another with the assumption, that the functions in use were part of the JACK API whereas they are actually defined and implemented in common/netjack_packet.{c,h}.

Both jack1 and jack2 can be compiled just fine against the current implementation.

File: tools/netsource.c (hunk 2 above)
Comment (dvzrv) [reply tools/netsource.c:39]: Has apparently been there since 2010 and is introduced by 02c76244.

Comment (dvzrv) [reply tools/netsource.c:39]: It is also still around in jack2: https://github.com/jackaudio/jack2/blob/a50aeb1f331546e86c1f35f26e67692c1cdfc9a5/tools/netsource.c#L43

Comment (dvzrv) [reply tools/netsource.c:39]: Do you want to apply a patch on top of this or add a ticket for it?

Comment (falkTX) [reply tools/netsource.c:39]: lets leave it as-is then, but it is very bad practice

**Pull request merged**